packages/blocks/src/store/process-block-type.js

```diff
@@ -197,5 +197,41 @@ export const processBlockType =
 			return;
 		}
 
+		if (
+			typeof settings?.parent === 'string' ||
+			settings?.parent instanceof String
+		) {
+			settings.parent = [ settings.parent ];
+			warning(
+				'Parent must be undefined or an array of strings (block types), but it is a string.'
+			);
+			// Intentionally continue:
+			//
+			// While string values were never supported, they appeared to work with some unintended side-effects
+			// that have been fixed by [#66250](https://github.com/WordPress/gutenberg/pull/66250).
+			//
+			// To be backwards-compatible, this code that automatically migrates strings to arrays.
+		}
+
+		if (
+			! Array.isArray( settings?.parent ) &&
+			settings?.parent !== undefined
+		) {
+			warning(
+				'Parent must be undefined or an array of block types, but it is ',
+				settings.parent
+			);
+			return;
+		}
+
+		if ( 1 === settings?.parent?.length && name === settings.parent[ 0 ] ) {
```

Comment:
I understand this check was carried from before, but why the check for `1`? Wouldn't the following be better in all respects?

```suggestion
		if ( settings?.parent?.includes?.( name ) ) {
```

Reply:
I'm not sure and it increases the surface area of this PR that we need to test, etc. — I'd rather do that separately. For reference, this is the [issue](https://github.com/WordPress/gutenberg/issues/66090) and [PR](https://github.com/WordPress/gutenberg/pull/66121).

Reply:
Valid point!